sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ThroughputRequestThrottler.java

```diff
@@ -126,6 +134,21 @@ private void updateAvailableThroughput() {
         }
     }
 
+    private static boolean isBulkRequest(RxDocumentServiceRequest request) {
```

Comment:
hmm, I think whether the bubble the 429 up probably should also do in ResourceThrottleRetryPolicy?  Then the same checking can be applied regardless whether throughput control is enabled.

Reply:
Attempted that - but we don't have sufficient context in the retry policies - we only get the exception - don't have access to the request (OperationType and RequestType). So this isn't possible without larger refactoring.